share/rocm/cmake/ROCMCreatePackage.cmake

```diff
@@ -11,32 +13,103 @@ set(ROCM_DEP_ROCMCORE TRUE CACHE BOOL "Add dependency on rocm-core package, defa
 include(CMakeParseArguments)
 include(GNUInstallDirs)
 include(ROCMSetupVersion)
+include(ROCMUtilities)
 
 find_program(MAKE_NSIS_EXE makensis)
 find_program(RPMBUILD_EXE rpmbuild)
 find_program(DPKG_EXE dpkg)
 
-if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.12.0")
-    # pretty much just a wrapper around string JOIN
-    function(rocm_join_if_set glue inout_variable)
-        string(JOIN "${glue}" to_set_parent ${${inout_variable}} ${ARGN})
-        set(${inout_variable} "${to_set_parent}" PARENT_SCOPE)
-    endfunction()
-else()
-    # cmake < 3.12 doesn't have string JOIN
-    function(rocm_join_if_set glue inout_variable)
-        set(accumulator "${${inout_variable}}")
-        set(tglue ${glue})
-        if(accumulator STREQUAL "")
-            set(tglue "")       # No glue needed if initially unset
+set(CMAKE_INSTALL_DEFAULT_COMPONENT_NAME runtime)
+set(ROCM_PACKAGE_CREATED FALSE CACHE INTERNAL "Track whether rocm_create_package has been called.")
+
+function(rocm_package_add_rpm_dependencies)
+    set(options QUIET)
+    set(oneValueArgs COMPONENT)
+    set(multiValueArgs DEPENDS SHARED_DEPENDS STATIC_DEPENDS)
+    cmake_parse_arguments(PARSE "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    if(${ROCM_PACKAGE_CREATED} AND NOT PARSE_QUIET)
```

Comment:
The `QUIET` flag suppresses the error message about the function being called after the package was created. It was put in so that the error messages from `rocm_package_add_rpm_dependencies` and `rocm_package_add_deb_dependencies` could be suppressed in favor of a more helpful single message from `rocm_package_add_dependencies`. I then added it to `rocm_package_add_dependencies` for consistency.